### ISSUE TITLE
Empty "email" property for a new Team

### ISSUE THREAD

**Issue by xmihux** [microsoftgraph/microsoft-graph-docs#3927]
If I GET a channel of a newly created Team using the beta API, the "email" property is empty, unless I physically go to the Team and click 'get e-mail address'. After that, the "email" property gets a value. Is there any workaround to get the email address automatically for a new team? The main reason I've been trying to do this is that chat messages to a channel are not supported with application permissions yet. Thanks!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0a639b2c-1646-2fb0-df4b-672717ecef06
* Version Independent ID: 4a5cc49a-ee65-02a5-583b-acfb0993a25f
* Content: [Get channel - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/channel-get?view=graph-rest-beta)
* Content Source: [api-reference/beta/api/channel-get.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/channel-get.md)
* Product: **microsoft-teams**
* GitHub Login: @nkramer
* Microsoft Alias: **jthake-msft**

**Comment by adamist521** [comment]
@xmihux 
Did you find a workaround for this?

**Comment by adamist521** [comment]
I know it's not supported yet (email property is `read-only`) but it said email `Type` was `Boolean` on the following doc so tried passing `"email" : true` on the request body at the creation of a channel but it failed with error attached.

https://docs.microsoft.com/en-us/graph/api/resources/channel?view=graph-rest-beta

```
{
    "error": {
        "code": "InvalidRequest",
        "message": "channel cannot be null.\r\nParameter name: channel",
        "innerError": {
            "request-id": "dd47b9ca-63b2-494c-a1d4-7138b4696148",
            "date": "2019-03-26T07:44:43"
        }
    }
}
```

**Comment by xmihux** [comment]
Not exactly. My current solution is to add a service/"bot" account to the
team, post message from as this user and then remove it from the team (from
flow). Not great but it works.

wt., 26 mar 2019, 15:49 użytkownik adamist521 <notifications@github.com>
napisał:

**Comment by Leif-Frederiksen** [comment]
I have the same issue - still hoping to hear from Microsoft if this is something that will be fixed in near future.

**Comment by holandaleo** [comment]
This issue has been reported back in Feb as beta api. But even as I'm writing this, the v.10 api has endpoints to [list all channels](https://docs.microsoft.com/en-us/graph/api/channel-list) or show [properties for specific channels](https://docs.microsoft.com/en-us/graph/api/channel-get). Both still hold the same original issue from beta version. Until some user click on UI's 'Get email address' I couldn't find any way to retrieve channel's email address straight from the API.

**Comment by ApoorvaPuppala** [comment]
Currently we are facing issue where the email address of the General Channel is empty (sometimes able to retrieve the channel email and some times unable to retrieve the channel email) using Microsoft Graph API. Below is the code i have written to 

using (HttpClient _httpClient = GethttpClient(token))
                {
                    int counter = 0;
                    var url = "https://graph.microsoft.com/v1.0/teams/" + teamid + "/channels";
                    do
                    {
                        HttpResponseMessage response = _httpClient.GetAsync(url).Result;
                        var content = response.Content.ReadAsStringAsync().Result;
                        var jsonData = JObject.Parse(content)["value"];
                        if (jsonData != null)
                        {
                            channelEmail = Convert.ToString(jsonData[0]["email"]);
                        }
                        counter++;
                    } while (counter <3);
                }

From the Teams App am able to get the email address by clicking on the Get email address.

Could anyone suggest how to overcome this issue?

**Comment by mohsinhere** [comment]
Did anyone find the issue? Been so long and I'm still having this issue.

**Comment by SubramanyamKirikera** [comment]
Hi Team, I am using Microsoft Graph API for creating the Channels. But After Creating , Email address of the Channel is showing as Empty.  Anybody has solution for this ?

**Comment by ryanps1** [comment]
Did anyone find a method to achieve this?

**Comment by nilsvbconxion** [comment]
Still having this issue on 21/09/2020. Any updates on this problem yet?

**Comment by khoa0712218** [comment]
It's 10/9/2020. Did anyone have any solution for this problem?

**Comment by j1151300783** [comment]
It's 10/23/2020. Anyone solution?
